democritus_dates/dates_and_times.py

```diff
@@ -459,3 +459,18 @@ def time_waste(n=3):
     time.sleep(n)
     message = f'I just wasted {n} seconds of your life.'
     print(message)
+
+def time_as_float(time_string: str) -> float:
+    """converts a given HH:MM time string to float"""
+    try:
+        hours, mintues = list(map(int, time_string.split(":"))) #parse given time string
+    except ValueError:
+        message = f"Invalid time string, ensure that the argument is in HH:MM format. Provided value: {time_string}"
+        raise ValueError(message)
```

Comment:
I suggest adding the suggestions I've made below. Adding `from e` to the end of the line where you raise a `ValueError` will make the traceback more clear if anyone does run into this exception.

*Before* adding `from e` the exception will look like:

```
{Traceback removed for brevity}
ValueError: invalid literal for int() with base 10: 'a'

During handling of the above exception, another exception occurred:

{Traceback removed for brevity}
ValueError: foo
```

*After* adding `from e` to the end of the `ValueError` call as suggested below will make the errors look like:

```
{Traceback removed for brevity}
ValueError: invalid literal for int() with base 10: 'a'

The above exception was the direct cause of the following exception:

{Traceback removed for brevity}
ValueError: foo
```

The relationship between the two ValueErrors is made more clear in the second case.

You can read more about the `raise ... from` construction [here](https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement).

```suggestion
    except ValueError as e:
        message = f"Invalid time string, ensure that the argument is in HH:MM format. Provided value: {time_string}"
        raise ValueError(message) from e
```